detectors/resources-support/src/main/java/com/google/cloud/opentelemetry/detectors/DetectedPlatform.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.opentelemetry.detectors;
+
+import java.util.Map;
+
+/** Represents a GCP specific platform on which a cloud application can run. */
+public interface DetectedPlatform {
+  /**
+   * Method to retrieve the underlying compute platform on which application is running.
+   *
+   * @return the {@link GCPPlatformDetector.SupportedPlatform} representing the Google Cloud
+   *     platform on which application is running.
+   */
+  GCPPlatformDetector.SupportedPlatform getSupportedPlatform();
+
+  /**
+   * Method to retrieve the attributes associated with the compute platform on which the application
+   * is running as key-value pairs.
+   *
+   * @return a {@link Map} of attributes specific to the underlying compute platform.
+   */
+  Map<String, String> getAttributes();
```

Comment:
It helped in writing simpler mocks and made the code testable. I don't think the overhead of additional constants is too much especially since some of them are getting re-used and Java string pool usually does a good job in re-using string literals. 

@jsuereth can maybe comment here on the trade-off.

Reply:
I'm not worried about the overhead, as resource attributes are a one-time cost.  Does this make it more complicated for the user of this library?

Reply:
The only user of this library is gonna be the upstream detector (detector-resources module) - The trade-off we make here is choosing between calling resource specific getter functions and making a get call on the Attributes map. 

While a getter function would be marginally simpler on the consumer side, it would mean exposing all the compute specific classes on the consumer side - since there would be no way to unify the individual GCP compute classes via an interface (since Getter methods would be different for each platform).

IMO, the code on the consumer side is not too complicated to read and maintain.

Reply:
ok, that sounds reasonable.